src/arrayMethodJoin.js

```diff
@@ -4,9 +4,31 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    if (this.length === 0) {
+      return '';
+    } else if (this.length === 1) {
```

Comment:
No need to include if (this.length === 1). If the length is 1, 2, 3 or more, you need to do the same things for it in your loop